yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -231,6 +249,17 @@ private[yarn] class YarnAllocator(
   }
 
   /**
+   * Creates a container request, handling the reflection required to use YARN features that were
+   * added in recent versions.
+   */
+  private def createContainerRequest(resource: Resource): ContainerRequest = {
+    nodeLabelConstructor.map { constructor =>
```

Comment:
I don't have a strong opinion on this either.  I don't think we've really set a precedence yet. Right now the code matches the documentation.  We could put it in as is (or change logInfo to logWarn) and if we find its an issue change it to error out.   @vanzin  thoughts?